gslab_scons/misc.py

```diff
@@ -342,9 +343,12 @@ def finder(rel_parent_dir, pattern, excluded_dirs = []):
     '''
 
     if is_unix():
-        command = 'find %s -name "%s" -type f' % (rel_parent_dir, pattern)
-        for x in excluded_dirs: # add in args to exclude folders from search
-            command = '%s -not -path "*%s*"' % (command, os.path.normpath(x)) 
+        command = 'find %s' % (rel_parent_dir)
+        if len(excluded_dirs) > 0:
+            for x in excluded_dirs: # add in args to exclude folders from search
+                command = '%s -path "*%s*" -prune -o ' % (command, os.path.normpath(x))

```

Comment:
Aren't you missing some `-not` switches and `\(`-type bracketing? I really can't tell.

It might be clearer to construct the base `command` and the `excluded_dirs` separately and then join them together in a third step. This method moves `command` around a lot, making it hard to keep track of.

Reply:
I agree it's super confusing (and quite frankly a miracle that it worked that well). Here's a breakdown. Suppose I want to find needle `*sconscript*.log` in haystack `.`, but want to avoid `build/data` and `build/external`.  The following two expressions are almost equivalent:

```
find . -path "*build/data*" -prune -o  -path "*build/external*" -prune -o  -name "*sconscript*.log" -type f
```

```
find . -not \( -path "*build/data*" -prune -o  -path "*build/external*" -prune \)  -name "*sconscript*.log" -type f
```

The first one says: in haystack `.`, prune `build/data` (i.e. if you hit `build/data`, skip and move on) OR prrune `build/external` OR search for needle. 

The second one (this is the one in the SO) says in hay stack `.`, if you did not prune (i.e. didn't hit) `build/data` AND did not prune `build/external` THEN you search for needle. 

I use the first one since it's a bit easier to understand for me.